src/navigation-instruction.js

```diff
@@ -161,19 +161,29 @@ export class NavigationInstruction {
   * Gets the instruction's base URL, accounting for wildcard route parameters.
   */
   getBaseUrl(): string {
+    let fragment = this.fragment;
+
+    if (fragment === '') {
+      let nonEmptyRoute = this.router.routes.find(route => {
+        return route.name === this.config.name &&
+          route.route !== '';
+      });
+      fragment = nonEmptyRoute.route;
```

Comment:
Do you mean there is a situation that supports an empty route fragment or that we should handle in case one is not met - 

```
if (!nonEmptyRoute) {
  return throw new Error('Matching route not found!');
}
fragment = nonEmptyRoute.route;
```